src/Entity/EntityId.php

```diff
@@ -17,6 +18,30 @@
 
 	protected $serialization;
 
+	const PATTERN = '/^:?(\w+:)*[^:]+\z/';
+
+	/**
+	 * @param string $serialization
+	 */
+	public function __construct( $serialization ) {
+		$this::assertValidSerialization( $serialization );
+		$this->serialization = self::normalizeIdSerialization( $serialization );
+	}
+
+	private static function assertValidSerialization( $serialization ) {
+		if ( !is_string( $serialization ) ) {
+			throw new InvalidArgumentException( '$serialization must be a string' );
+		}
+
+		if ( $serialization === '' ) {
```

Comment:
I will always merge it when I see it. Super trivial and super easy to understand and to read. I wish there would be a "non-empty string" check in Assert. This pattern repeats a thousand times in our code base.